Linux is special, platform is linux2 even for linux-3.x kernels...

--- configure~	2011-12-02 18:50:18.950958621 +0100
+++ configure	2011-12-02 18:53:35.396488967 +0100
@@ -2961,6 +2961,7 @@
 	darwin*) MACHDEP="darwin";;
 	atheos*) MACHDEP="atheos";;
         irix646) MACHDEP="irix6";;
+        linux*)  MACHDEP="linux2";;
 	'')	MACHDEP="unknown";;
 	esac
 fi
